### DISCUSSION TITLE
If building with two toolchain (cl and mingw) how to distinguish packages?

### DISCUSSION THREAD

**Discussion by DaddyXtof** [xmake-io/xmake#3439] [Q&A] [answered]
If I am building a Windows application and I have defined two targets (one to be built built with **cl** and the other with **mingw/gcc**) and I have a dependency on a package (raylib) then the build for mingw will fail as it tries to link with the raylib library built with cl.

This is my xmake.lua:
```
set_defaultplat("windows")
add_requires("raylib","raygui")
add_rules("mode.debug", "mode.release")

target("windows")
    set_filename("raygui-scaffold-win.exe")
    set_plat("windows")
    add_packages("raylib","raygui")
    set_kind("binary")
    add_files("src/*.c")

target("mingw")
    set_filename("raygui-scaffold-mingw.exe")
    set_plat("mingw")
    set_toolchains("mingw")
    add_packages("raylib","raygui")
    set_kind("binary")
    add_files("src/*.c")
```
In order to make the mingw target work I have to explicitly a **add_linkdir** pointing to the lib built with the other toolchain.
Am i missing something or is this a limitation of how packages are downloaded and built?
If you have the same package dependency but for two different toolchains how do you distinguish them in the xmake.lua?

**Comment by waruqi** [comment, marked as answer]
```
add_requires("raylib")
add_requires("raylib~mingw", {plat = "mingw", arch = "x86_64"})
```

**Reply by DaddyXtof** [reply]
Thank you for the quick reply - when using add_requires("raylib~mingw", {plat = "mingw", arch = "x86_64"})
I get 
```
error: method 'fetch' is not callable (a nil value)
  => install raylib~mingw 4.2.0 .. failed
error: install failed!
```
Am on v2.7.7

**Reply by DaddyXtof** [reply]
and for clarity this is my current xmake.lua:
```
add_requires("raylib","raygui")
add_requires("raylib~mingw", {plat = "mingw", arch = "x86_64"})
add_rules("mode.debug", "mode.release")

target("windows")
    set_filename("raygui-scaffold-win.exe")
    set_plat("windows")
    add_packages("raylib","raygui")
    set_kind("binary")
    add_files("src/*.c")

target("mingw")
    set_filename("raygui-scaffold-mingw.exe")
    set_plat("mingw")
    set_toolchains("mingw")
    add_packages("raylib~mingw","raygui")
    set_kind("binary")
    add_files("src/*.c")

**Reply by waruqi** [reply]
try it again. https://github.com/xmake-io/xmake/pull/3442

xmake update -s dev

**Reply by DaddyXtof** [reply]
Excellent - it works.
Thank you for the great tool, and fast and efficient support!
